bindings/python/pymongoarrow/api.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2021-present MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import warnings
+
+from pymongoarrow.context import PyMongoArrowContext
+from pymongoarrow.lib import process_bson_stream
+
+
+def find_arrow_all(collection, query, *, schema, **kwargs):
+    """Method that returns the results of a find query as a
+    :class:`pyarrow.Table` instance.
+
+    :Parameters:
+      - `collection`: Instance of :class:`~pymongo.collection.Collection`.
+        against which to run the ``find`` operation.
+      - `query`: A mapping containing the query to use for the find operation.
+      - `schema`: Instance of :class:`~pymongoarrow.schema.Schema`.
+
+    Additional keyword-arguments passed to this method will be passed
+    directly to the underlying ``find`` operation.
+
+    :Returns:
+      An instance of class:`pyarrow.Table`.
+    """
+    context = PyMongoArrowContext.from_schema(schema)
+
+    for opt in ('session', 'cursor_type', 'session', 'projection'):
+        if kwargs.pop(opt, None):
+            warnings.warn(
+                'Ignoring option {!r} as it is not supported by '
+                'PyMongoArrow'.format(opt), UserWarning, stacklevel=2)
```

Comment:
Is there any particular reason we're using `"".format()` instead of format strings like `f'..{opt}...'`?

Reply:
None other than I wasn't fully familiar with f-strings. Changed :)